src/lib/types.ts

```diff
@@ -21,36 +25,129 @@ export type CCIPMessage = AbiParametersToPrimitiveTypes<
 export type CCIPContractTypeOnRamp = typeof CCIPContractTypeOnRamp
 export const CCIPContractTypeOffRamp = 'EVM2EVMOffRamp'
 export type CCIPContractTypeOffRamp = typeof CCIPContractTypeOffRamp
 export const CCIPContractTypeCommitStore = 'EVM2EVMCommitStore'
 export type CCIPContractTypeCommitStore = typeof CCIPContractTypeCommitStore
+export const CCIPContractTypeBurnMintTokenPool = 'BurnMintTokenPool'
+export type CCIPContractTypeBurnMintTokenPool = typeof CCIPContractTypeBurnMintTokenPool
+export const CCIPContractTypeBurnMintTokenPoolAndProxy = 'BurnMintTokenPoolAndProxy'
+export type CCIPContractTypeBurnMintTokenPoolAndProxy =
+  typeof CCIPContractTypeBurnMintTokenPoolAndProxy
+export const CCIPContractTypeBurnMintTokenPoolWithTax = 'BurnMintTokenPoolWithTax'
+
+export const CCIPContractTypeBurnWithFromMintTokenPool = 'BurnWithFromMintTokenPool'
+export type CCIPContractTypeBurnWithFromMintTokenPool =
+  typeof CCIPContractTypeBurnWithFromMintTokenPool
+
+export const CCIPContractTypeBurnWithFromMintTokenPoolAndProxy = 'BurnWithFromMintTokenPoolAndProxy'
+export type CCIPContractTypeBurnWithFromMintTokenPoolAndProxy =
+  typeof CCIPContractTypeBurnWithFromMintTokenPoolAndProxy
+
+export type CCIPContractTypeBurnMintTokenPoolWithTax =
+  typeof CCIPContractTypeBurnMintTokenPoolWithTax
+export const CCIPContractTypeBurnWithFromMintRebasingTokenPool = 'BurnWithFromMintRebasingTokenPool'
+export type CCIPContractTypeBurnWithFromMintRebasingTokenPool =
+  typeof CCIPContractTypeBurnWithFromMintRebasingTokenPool
+export const CCIPContractTypeLockReleaseTokenPool = 'LockReleaseTokenPool'
+export type CCIPContractTypeLockReleaseTokenPool = typeof CCIPContractTypeLockReleaseTokenPool
+export const CCIPContractTypeLockReleaseTokenPoolAndProxy = 'LockReleaseTokenPoolAndProxy'
+export type CCIPContractTypeLockReleaseTokenPoolAndProxy =
+  typeof CCIPContractTypeLockReleaseTokenPoolAndProxy
+export const CCIPContractTypeUSDCTokenPool = 'USDCTokenPool'
+export type CCIPContractTypeUSDCTokenPool = typeof CCIPContractTypeUSDCTokenPool
+
+export const CCIPContractTypeTokenPool = [
+  CCIPContractTypeBurnMintTokenPool,
+  CCIPContractTypeBurnMintTokenPoolAndProxy,
+  CCIPContractTypeBurnMintTokenPoolWithTax,
+  CCIPContractTypeBurnWithFromMintRebasingTokenPool,
+  CCIPContractTypeLockReleaseTokenPool,
+  CCIPContractTypeLockReleaseTokenPoolAndProxy,
+  CCIPContractTypeUSDCTokenPool,
+  CCIPContractTypeBurnWithFromMintTokenPoolAndProxy,
+  CCIPContractTypeBurnWithFromMintTokenPool,
+]
+export type CCIPContractTypeTokenPool =
+  | CCIPContractTypeBurnMintTokenPool
+  | CCIPContractTypeLockReleaseTokenPool
+  | CCIPContractTypeLockReleaseTokenPoolAndProxy
+  | CCIPContractTypeUSDCTokenPool
+  | CCIPContractTypeBurnWithFromMintRebasingTokenPool
+  | CCIPContractTypeBurnMintTokenPoolAndProxy
+  | CCIPContractTypeBurnMintTokenPoolWithTax
+  | CCIPContractTypeBurnWithFromMintTokenPoolAndProxy
+  | CCIPContractTypeBurnWithFromMintTokenPool
```

Comment:
Do we really need these contract types here?
When I created them, it was with the assumption that token pools were not strictly part of the ccip core contracts. They can even be implemented by 3Ps, and not comply with our interfaces at all.
I'd rather leave them to some self-contained command, and not tie them directly into lib.